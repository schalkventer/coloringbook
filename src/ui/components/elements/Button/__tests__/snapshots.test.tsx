import React from 'react';
import renderer from 'react-test-renderer';
import { Default, Actions, Colors, Disabled, Icon, Loading } from './stories';

test('Demo', (): void => expect(renderer.create(<Default />).toJSON()).toMatchSnapshot());
test('Actions', (): void => expect(renderer.create(<Actions />).toJSON()).toMatchSnapshot());
test('Colors', (): void => expect(renderer.create(<Colors />).toJSON()).toMatchSnapshot());
test('Disabled', (): void => expect(renderer.create(<Disabled />).toJSON()).toMatchSnapshot());
test('Icon', (): void => expect(renderer.create(<Icon />).toJSON()).toMatchSnapshot());
test('Loading', (): void => expect(renderer.create(<Loading />).toJSON()).toMatchSnapshot());
