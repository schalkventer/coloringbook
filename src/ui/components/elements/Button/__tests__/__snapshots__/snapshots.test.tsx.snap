// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions 1`] = `
Array [
  <div
    aria-describedby={null}
    className="sc-htoDjs cvnhzW"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title=""
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bZQynM cRWftn"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      swatch="grey"
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span>
          Hello World!
        </span>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>,
  <div
    aria-describedby={null}
    className="sc-htoDjs cvnhzW"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title=""
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bZQynM cRWftn"
      href="/example"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      swatch="grey"
      tabIndex={0}
      target="_self"
    >
      <span
        className="MuiButton-label"
      >
        <span>
          Hello World!
        </span>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </a>
  </div>,
  <div
    aria-describedby={null}
    className="sc-htoDjs cvnhzW"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title=""
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bZQynM cRWftn"
      href="https://www.google.com"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      swatch="grey"
      tabIndex={0}
      target="_blank"
    >
      <span
        className="MuiButton-label"
      >
        <span>
          Hello World!
        </span>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </a>
  </div>,
]
`;

exports[`Colors 1`] = `
Array [
  <div
    aria-describedby={null}
    className="sc-htoDjs cvnhzW"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title=""
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bZQynM cRWftn"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      swatch="grey"
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span>
          Hello World!
        </span>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>,
  <div
    aria-describedby={null}
    className="sc-htoDjs cvnhzW"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title=""
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bZQynM fUzIXl"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      swatch="blue"
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span>
          Hello World!
        </span>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>,
]
`;

exports[`Demo 1`] = `
<div
  aria-describedby={null}
  className="sc-htoDjs cvnhzW"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title=""
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bZQynM cRWftn"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    swatch="grey"
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <span>
        Hello World!
      </span>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Disabled 1`] = `
<div
  aria-describedby={null}
  className="sc-htoDjs cvnhzW"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title="The reason why this is disabled"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bZQynM hnQJdt Mui-disabled Mui-disabled"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    swatch="grey"
    tabIndex={-1}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <span>
        Hello World!
      </span>
    </span>
  </button>
</div>
`;

exports[`Icon 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    aria-describedby={null}
    className="sc-htoDjs cvnhzW"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title=""
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bZQynM cRWftn"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      swatch="grey"
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          style={
            Object {
              "height": "24px",
              "opacity": 1,
              "width": "24px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM17 13l-5 5-5-5h3V9h4v4h3z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    aria-describedby={null}
    className="sc-htoDjs cvnhzW"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title=""
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bZQynM cRWftn"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      swatch="grey"
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          style={
            Object {
              "height": "13px",
              "opacity": 1,
              "width": "13px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM17 13l-5 5-5-5h3V9h4v4h3z"
          />
        </svg>
        <span
          className="sc-gzVnrw cDMSXN"
        >
          Hello World!
        </span>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    aria-describedby={null}
    className="sc-htoDjs cvnhzW"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title=""
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bZQynM cRWftn"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      swatch="grey"
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span
          className="sc-gzVnrw cDMSXN"
        >
          Hello World!
        </span>
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          style={
            Object {
              "height": "13px",
              "opacity": 1,
              "width": "13px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM17 13l-5 5-5-5h3V9h4v4h3z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Loading 1`] = `
Array [
  <div
    aria-describedby={null}
    className="sc-htoDjs cvnhzW"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title=""
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bZQynM hnQJdt Mui-disabled Mui-disabled"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      swatch="grey"
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span>
          Hello World!
        </span>
        <div
          className="sc-dnqmqq Fbgdo"
        >
          <div
            className="MuiCircularProgress-root sc-iwsKbI irMHdh MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style={
              Object {
                "height": 18,
                "width": 18,
              }
            }
            swatch="grey"
          >
            <svg
              className="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx={44}
                cy={44}
                fill="none"
                r={20.2}
                strokeWidth={3.6}
                style={Object {}}
              />
            </svg>
          </div>
        </div>
      </span>
    </button>
  </div>,
  <div
    aria-describedby={null}
    className="sc-htoDjs cvnhzW"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title=""
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bZQynM cRWftn"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      swatch="grey"
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span>
          Hello World!
        </span>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>,
  <div
    aria-describedby={null}
    className="sc-htoDjs cvnhzW"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title=""
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bZQynM cRWftn"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      swatch="grey"
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span>
          Hello World!
        </span>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>,
]
`;
